src/main/java/Duke.java

```diff
@@ -1,10 +1,289 @@
+            if (inputString.equals("bye")) {
+                saveDataToFile("tasks", tasks);
+                exit();
+                break;
+            } else if (inputString.equals("list")) {
+                try {
+                    printList(tasks);
+                } catch (DukeException d) {
+                    System.out.println(line
+                            + d.getMessage() + "\n" + line);
+                }
+            } else if (inputString.contains("done") == true) {
+                try {
+                    setDone(inputString, tasks);
+                } catch (IndexOutOfBoundsException i) {
+                    System.out.println(line
+                            + "☹ OOPS!!! The index is out of range.\n"
+                            + line);
+                } catch (NumberFormatException n) {
+                    System.out.println(line
+                            + "☹ OOPS!!! The index of done indstruction should be a number.\n"
+                            + line);
+                } catch (NullPointerException p) {
+                    System.out.println(line
+                            + "☹ OOPS!!! The index of done indstruction cannot be empty.\n"
+                            + line);
+                }
+            } else if (inputString.contains("delete") == true) {
+                try {
+                    deleteTask(inputString, tasks);
+                } catch (IndexOutOfBoundsException i) {
+                    System.out.println(line
+                            + "☹ OOPS!!! The index is out of range.\n"
+                            + line);
+                } catch (NumberFormatException n) {
+                    System.out.println(line
+                            + "☹ OOPS!!! The index of delete indstruction should be a number.\n"
+                            + line);
+                }
+            } else if (inputString.contains("clear") == true) {
+                clearList(tasks);
+            } else {
+                try {
+                    addTask(inputString, tasks);
+                }
+                catch(IndexOutOfBoundsException i) {
+                    String type = "";
+                    if (inputString.contains("todo")) {
+                        type = "todo";
+                    } else if (inputString.contains("deadline")) {
+                        type = "deadline";
+                    } else if (inputString.contains("event")) {
+                        type = "event";
+                    }
+                    if (type == "") {
+                        System.out.println(line
+                                + "☹ OOPS!!! I'm sorry, but I don't know what that means :-(\n"
+                                + line);
+                    } else {
+                        System.out.println(line
+                                + "☹ OOPS!!! The description of a " + type + " cannot be empty.\n"
+                                + line);
+                    }
```

Comment:
Avoid using so many nested try catch blocks. Please abstract out the logic into different functions